modules/ui/rapid_feature_toggle_dialog.js

```diff
@@ -0,0 +1,177 @@
+        var drawAiFeatures = context.layers().layer('ai-features');
+        drawAiFeatures.enabled(!drawAiFeatures.enabled()); 
+        redrawOnToggle(); 
+    }
+
+
+    function keyPressFormHandler(){
+        if (d3_event.shiftKey && 
+            d3_event.key === 'R'){
+            handleToggleAllClick(); 
+        }
+    }
+
+
+    function redrawOnToggle() {
+        var drawAiFeatures = context.layers().layer('ai-features');
+        // We need check/uncheck the 'all options' boxes, and 
+        // disable the other checkboxes so that the user
+        // cannot interact with them. 
+        var roadCheckbox = d3_select('#road-toggle');
+        var buildingCheckbox = d3_select('#building-toggle');
+        var allCheckbox = d3_select('#all-toggle');
+
+        // We also need to add a class to the whole option so that we 
+        // can style it accordingly. 
+        var roadOption = d3_select('#section-road-toggle');
+        var buildingOption = d3_select('#section-building-toggle');
+
+        if (drawAiFeatures.showAll()) {
+            allCheckbox.property('checked', true); 
+            roadCheckbox.attr('disabled', null);  
+            buildingCheckbox.attr('disabled', null);
+            roadOption.classed('disabled', false); 
+            buildingOption.classed('disabled', false); 
+            
+        } else {
+            allCheckbox.property('checked', false); 
+            roadCheckbox.attr('disabled', true); 
+            buildingCheckbox.attr('disabled', true); 
+            roadOption.classed('disabled', true); 
+            buildingOption.classed('disabled', true); 
+        }
+    }
+
+
+    return function(selection) {
+        var modalSelection = uiModal(selection);
+
+        modalSelection.select('.modal')
+            .attr('class', 'modal-splash modal modal-rapid');
+
+        var modal = modalSelection.select('.content')
+            .append('div')
+            .attr('class', 'fillL feature stack');
+
+
+        var drawAiFeatures = context.layers().layer('ai-features');
+
+        addCheckBox(modal, 
+            'all-toggle',
+            'rapid_feature_toggle.toggle_all', 
+            null,
+            handleToggleAllClick, 
+            drawAiFeatures.showAll(),
+            false
+            ); 
+    
+        modal
+            .append('div')
+            .attr('class','modal-section rapid-checkbox section-divider');
+
+        addCheckBox(modal,
+            'road-toggle',
+            'rapid_feature_toggle.roads', 
+            'rapid_feature_toggle.roads_provided_by',
+            toggleSvgRoads,
+            drawAiFeatures.showRoads(),
+            !drawAiFeatures.showAll()
+            ); 
+
+        addCheckBox(modal, 
+            'building-toggle',
+            'rapid_feature_toggle.buildings', 
+            'rapid_feature_toggle.buildings_provided_by',
+            toggleSvgBuildings,
+            drawAiFeatures.showBuildings(),
+            !drawAiFeatures.showAll()
+            ); 
```

Comment:
This works, but it's tough to read because of all the options. An options object could improve it as the sole argument:
```
addCheckBox({
  modal: modal,
  id: 'building-toggle',
  label: 'rapid_feature_toggle.buildings'
  description: 'rapid_feature_toggle.buildings_provided_by'
  handler: toggleSvgBuildings
  enabled: drawAiFeatures.showBuildings()
  greyout: !drawAiFeatures.showAll()
})

Reply:
Better to pass actual values for labels and description to avoid hiding the complexity: `label: t('rapid_feature_toggle.buildings')`

Sometimes rapidicon gets replaced, which can be handled by `label: t('rapid_feature_toggle.buildings', subsitutions)` easily enough to avoid repeating yourself.

Reply:
Yep, this is a cleaner separation of concerns and less overload-y.